### PR TITLE
Bump to mono/2018-04

### PR DESCRIPTION
**Pull request by luhenry**

### CODE REVIEW

**Comment by atsushieno** [comment]
should we keep this open?

**Comment by luhenry** [comment]
@atsushieno yes

**Comment by jonpryor** [comment]
On the plus side, it *built* this time.  (Yay!)

On the minus side, we have [26 unit test failures](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/3614/testReport/).

Progress!

**Comment by marek-safar** [comment]
@jonp there are 0 test failures now

**Comment by marek-safar** [comment]
build

**Comment by marek-safar** [comment]
build

**Comment by marek-safar** [comment]
build

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 1]
```diff
@@ -307,9 +307,8 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
 	<_InstantRunEnabled Condition=" '$(_InstantRunEnabled)' == '' ">False</_InstantRunEnabled>
 	<_AndroidBuildPropertiesCache>$(IntermediateOutputPath)build.props</_AndroidBuildPropertiesCache>
 	<_AndroidDesignTimeBuildPropertiesCache>$(_AndroidIntermediateDesignTimeBuildDirectory)build.props</_AndroidDesignTimeBuildPropertiesCache>
-
 	<AndroidGenerateJniMarshalMethods Condition=" '$(AndroidGenerateJniMarshalMethods)' == '' ">False</AndroidGenerateJniMarshalMethods>
-
+	<MakeBundleKeepTemporaryFiles Condition=" '$(MakeBundleKeepTemporaryFiles)' == '' ">False</MakeBundleKeepTemporaryFiles>
```
[the remark this answers is not in the document]
Comment (jonpryor) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets]: I mean that MSBuild properties that we use should either have a `_` prefix (for "private" use) or should have an `Android` prefix (for "public" use), e.g.

```xml
<AndroidMakeBundleKeepTemporaryFiles Condition=" '$(AndroidMakeBundleKeepTemporaryFiles)' == '' ">False</AndroidMakeBundleKeepTemporaryFiles>
```

**Pull request merged**